src/Resources/public/js/search.js

```diff
@@ -43,43 +43,111 @@ class SearchManager {
+            }
+        });
+
+        this.#initializeForm();
+    }
+
+    #initializeForm() {
         let form = this.#options.form;
 
-        if(typeof form === 'string') {
+        if (typeof form === 'string') {
             form = document.querySelector(form);
 
-            if(null === form) {
+            if (null === form) {
                 throw new Error('Form not found');
             }
         }
 
         this.#form = form;
 
-        this.#form.addEventListener('input', function (event) {
-            /** @type {HTMLInputElement} */
+        this.#form.addEventListener('input', (event) => {
             const field = event.target;
 
+            if (!(field instanceof HTMLInputElement || field instanceof HTMLSelectElement)) {
+                return;
+            }
+
             field.dispatchEvent(new CustomEvent('search:form-changed', { bubbles: true }));
 
             // todo would be better to have a way to check the type of the field directly on the field, e.g. with a data attribute
-            if(field.closest('.ssm-filter') !== null) {
+            if (field.closest('.ssm-filter') !== null) {
                 field.dispatchEvent(new CustomEvent('search:filter-changed', { bubbles: true }));
-            } else if(field.name === 'p') {
+            } else if (field.name === 'p') {
                 field.dispatchEvent(new CustomEvent('search:page-changed', { bubbles: true }));
-            } else if(field.classList.contains('ssm-sort')) {
+            } else if (field.classList.contains('ssm-sort')) {
                 field.dispatchEvent(new CustomEvent('search:sort-changed', { bubbles: true }));
             }
         });
 
-        this.#form.addEventListener('search:filter-changed', (event) => this.#options.onFilterChange.bind(this, this.#form, event.target)());
-        this.#form.addEventListener('search:page-changed', (event) => this.#options.onSortChange.bind(this, this.#form, event.target)());
-        this.#form.addEventListener('search:sort-changed', (event) => this.#options.onSortChange.bind(this, this.#form, event.target)());
-        this.#form.addEventListener('submit', () => this.#options.onSubmit.bind(this, this.#form)());
+        this.#form.addEventListener('search:filter-changed', (event) => this.#options.onFilterChange(this.#form, event.target));
+        this.#form.addEventListener('search:page-changed', (event) => this.#options.onPageChange(this.#form, event.target));
+        this.#form.addEventListener('search:sort-changed', (event) => this.#options.onSortChange(this.#form, event.target));
+        this.#form.addEventListener('search:update-url', (event) => this.#options.onUpdateUrl(this.#form, event.target));
+        this.#form.addEventListener('submit', (event) => {
+            event.preventDefault();
+            this.#options.onSubmit(this.#form);
+            this.#form.dispatchEvent(new CustomEvent('search:update-url', { bubbles: true }));
```

Comment:
Now looking at this I would rather it was in the `this.#options.onSubmit()` method. If people want to override that method, they can update the URL if they need to

Reply:
This way you can change one without the other.